cdap-api/src/main/java/co/cask/cdap/internal/ProgramType.java

```diff
@@ -0,0 +1,133 @@
+/*
+ * Copyright © 2014 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.cdap.internal;
+
+import co.cask.cdap.api.schedule.SchedulableProgramType;
+import com.google.gson.annotations.SerializedName;
+
+import java.util.HashMap;
+import java.util.Map;
+import javax.annotation.Nullable;
+
+/**
+ * Defines types of programs supported by the system.
+ */
+public enum ProgramType {
+
+  // @SerializedName to maintain backwards-compatibility
+
+  @SerializedName("Flow")
+  FLOW(1, "flows", "Flow", true),
+
+  @SerializedName("Procedure")
+  PROCEDURE(2, "procedures", "Procedure", true),
+
+  @SerializedName("Mapreduce")
+  MAPREDUCE(3, "mapreduce", "MapReduce", true, SchedulableProgramType.MAPREDUCE),
+
+  @SerializedName("Workflow")
+  WORKFLOW(4, "workflows", "Workflow", true, SchedulableProgramType.WORKFLOW),
+
+  @SerializedName("Webapp")
+  WEBAPP(5, "webapp", "Webapp", false),
+
+  @SerializedName("Service")
+  SERVICE(6, "services", "Service", true),
+
+  @SerializedName("Spark")
+  SPARK(7, "spark", "Spark", true, SchedulableProgramType.SPARK),
+
+  @SerializedName("Worker")
+  WORKER(8, "workers", "Worker", true),
+
+  CUSTOM_ACTION(9, "custom", "Custom", false, SchedulableProgramType.CUSTOM_ACTION);
```

Comment:
I checked with @sagarkapare. Seems like there isn't a usage. @jwang47 Can you confirm?